packages/core/src/enqueue_links/enqueue_links.ts

```diff
@@ -128,17 +128,30 @@ export enum EnqueueStrategy {
 
     /**
      * Matches any URLs that have the same hostname.
-     * For example, `https://wow.example.com/hello` will be matched for a base url of `https://example.com/`, but
+     * For example, `https://wow.example.com/hello` will be matched for a base url of `https://wow.example.com/`, but
      * `https://example.com/hello` will not be matched.
+     *
+     * > This strategy will match both `http` and `https` protocols regardless of the base URL protocol.
      */
     SameHostname = 'same-hostname',
 
     /**
-     * Matches any URLs that have the same (sub-)domain as the base URL.
+     * Matches any URLs that have the same domain as the base URL.
      * For example, `https://wow.an.example.com` and `https://example.com` will both be matched for a base url of
      * `https://example.com`.
+     *
+     * > This strategy will match both `http` and `https` protocols regardless of the base URL protocol.
      */
     SameDomain = 'same-domain',
```

Comment:
Let's explain what is a "domain" here, because it really wasn't obvious to me. According to `tldts` it's `example.com` or `bbc.co.uk`, but not `foo.example.com` or `example.co.com`. So `foo.example.com` and `example.com` would match both same-domain rule on `hello.example.com` (and the same with `bbc.co.uk` instead of `example.com`).

Also, note that this really only applies to `tld`s, but not various different public suffixes, such as `github.io` (`foo.github.io` and `bar.github.io` are same-domain, according to this) - if we would want to fix that, we could use `getPublicSuffix` instead of `getDomain`

Reply:
> Also, note that this really only applies to `tld`s, but not various different public suffixes, such as `github.io` (`foo.github.io` and `bar.github.io` are same-domain, according to this) - if we would want to fix that, we could use `getPublicSuffix` instead of `getDomain`

I don't fully understand what you mean by this... That is the correct match for SameDomain given your examples 👀

Reply:
> mvolfik.github.io and apify.github.io/crawlee are same-domain, according to this

That library is a bit weird when it comes to naming, in our terms (and we were iterating on this several times already, so I hope they are correct), those should match same-hostname (github.io) but not same-domain, as the subdomain differs (and that is part of the domain). Note that we refer to just "domain", not "domain name" (which would be just "github" here), nor "tld/top-level domain" (".io" here) - for me the term "domain" consists of all the domain parts, hence it includes the subdomain part too.

Agreed we should explicitly document what the terminology here means. Ideally in the property description in the options interface:

https://crawlee.dev/api/core/interface/EnqueueLinksOptions#strategy

Reply:
It's the other way, @B4nan, same-hostname would also match the subdomain too, same-domain would match just the domain. 
![image](https://user-images.githubusercontent.com/17960496/214307695-a6fb8af6-beea-452a-bcb8-bef6d19e3a05.png)

Agreed that we definitely need to explain this, with visible examples, to users who might also be confused by the terminology

Reply:
Haha, glad B4nan just made my confusion even more justified :sweat_smile: 



> > Also, note that this really only applies to `tld`s, but not various different public suffixes, such as `github.io` (`foo.github.io` and `bar.github.io` are same-domain, according to this) - if we would want to fix that, we could use `getPublicSuffix` instead of `getDomain`
> 
> I don't fully understand what you mean by this... That is the correct match for SameDomain given your examples eyes

In a way... yes. But `github.io` is something like `co.uk`, just private (and there are many such private "domain zones", see [Mozilla's public suffix list](https://publicsuffix.org/list/)). We can definitely keep it like this, I just wanted to bring this to your attention.

Reply:
![image](https://user-images.githubusercontent.com/615580/214320585-89d4bbee-ba8d-43be-b211-fd94b8354b01.png)